extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx

```diff
@@ -1,15 +1,112 @@
 import * as React from 'react';
 import * as Rdom from 'react-dom';
+import * as octicons from '../../view/octicons';
 
 import { vscode } from '../../view/vscode-api';
 
+interface AnalysisResult {
+  nwo: string,
+  resultCount: number,
+  downloadLink: string,
+  fileSize: string,
+}
+
+interface Props {
+  queryTitle: string;
+  queryFile: string;
+  totalRepositoryCount: number;
+  totalResultCount: number;
+  executionTimestamp: string;
+  executionDuration: string;
+  downloadLink: string;
+  results: AnalysisResult[]
+}
+
+const AnalysisResult = (props: AnalysisResult) => (
+  <span>
+    <span className="analysis-item">{octicons.repo}</span>
+    <span className="analysis-item">{props.nwo}</span>
+    <span className="analysis-item badge-container">
+      <span className="badge">{props.resultCount}</span>
+    </span>
+    <span className="analysis-item">
+      <a
+        className="download-link"
+        href={props.downloadLink}>
+        {octicons.download}{props.fileSize}
+      </a>
```

Comment:
So, is the idea for the webview to make the requests directly? Or will it pass a message to the extension to make the request for the webview?

For now, this doesn't need to be worked out completely, but this is going to be an important decision to make. Some things to think about:

1. If the webview makes the request directly, we need to make sure the CSP allows it to happen (not difficult to do).
2. If the webview makes the request, then we need to figure out how to cache the data between runs. Does it send the data to the extension through a message and the extension saves it? Or does the webview save it directly (I think this is possible, but need to figure out exactly how the webview can access the file system).

My opinion is that _initially_ it will be simpler to make the request directly through the webview, but it will make things more complex later on since the webview still needs to communicate with the extension for caching and displaying query contents in the query history view.

Whatever decision we make initially, let's try to do it in a way that it's not difficult to change.

Reply:
In general, my thinking was that it would keep things simpler if the WebView would communicate with the rest of the world via the extension engine. But I completely forgot about it here, so you make a really good point!

I think we should leave things as is for now since this is a mock view, but when it comes to it, those links can turn into buttons that communicate with the extension that manages the download.